src/org/wojcieszko/tree/MyTree.java

```diff
@@ -0,0 +1,134 @@
+package org.wojcieszko.tree;
+
+public class MyTree<Integer> {
+    public MyTreeNode<Integer> root;
+
+    public MyTree(Integer data) {
+        root = new MyTreeNode<>(data);
+        root.setLeftLeaf(null);
+        root.setRightLeaf(null);
+
+    }
+
+    public boolean hasLeftLeaf(MyTreeNode myTreeNode) {
+        if (myTreeNode.getLeftLeaf() == null) {
+            return false;
+        } else return true;
+    }
+
+    public boolean hasRightLeaf(MyTreeNode myTreeNode) {
+        if (myTreeNode.getRightLeaf() == null) {
+            return false;
+        } else return true;
+    }
+
+    public Compare compareDataValue(MyTreeNode first, MyTreeNode second) {
+
+        if ((int) first.getData() < (int) second.getData()) {
+            return Compare.BIGGER;
+        } else if ((int) first.getData() == (int) second.getData()) {
+            return Compare.EQUAL;
+        } else if ((int) first.getData() > (int) second.getData()) {
+            return Compare.SMALLER;
+        } else return null;
+
+    }
+
+
+    public void appendLeaf(MyTreeNode tmpRoot, MyTreeNode myLeaf) {
+
+        myLeaf.setRoot(tmpRoot);
```

Comment:
Sporo komentarzy wywaliłem po przeczytaniu kodu dalej:

_root_ to nie jest root, a _parent_ (!?). Sporo zmienia.